src/ssr-template-helper.js

```diff
@@ -113,34 +107,36 @@ export class SsrTemplateHelper {
    */
   buildPayload_(
     request, mustacheTemplate, opt_templates, opt_attributes = {}) {
-    const ampComponent = dict({
-      'type': this.sourceComponent_,
-      'successTemplate': {
+    const ampComponent = dict({'type': this.sourceComponent_});
+    if ((opt_templates && opt_templates['successTemplate'])
+        || mustacheTemplate) {
+      ampComponent['successTemplate'] = {
         'type': 'amp-mustache',
-        'payload': opt_templates
-          ? this.xmls_.serializeToString(opt_templates['successTemplate'])
+        'payload': (opt_templates && opt_templates['successTemplate'])
+          ? opt_templates['successTemplate']./*REVIEW*/innerHTML
           : mustacheTemplate,
-      },
-      'errorTemplate': {
+      };
+    }
+    if (opt_templates && opt_templates['errorTemplate']) {
```

Comment:
ditto (less important to do it in this case, but it makes the code more consistent)

Reply:
Done.